[Kang Qiao] iP

kangqiao322 opened the pull request:
# Duke💀

>“Your mind is for having ideas, not holding them.” – David Allen 

Duke frees your mind of having to remember things you need to do. It's

- text-based
- easy to learn(maybe)
- SUPER FAST to use

All you need to do is,

1. download Duke from [here](https://github.com/kangqiao322/ip).
2. run it.
3. add your tasks.

and let it manage your tasks for you!🤓🤮 
And it is FREE!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(Duke.class, args);
    }
}
```

Commit:
Add Gradle support

kangqiao322 committed:
Level-1, Greet, Echo, Exit

kangqiao322 committed:
Update Duke.java

kangqiao322 committed:
Update Duke.java

kangqiao322 committed:
Level-2. Add, List

kangqiao322 committed:
Update Duke.java

kangqiao322 committed:
Level-2

kangqiao322 committed:
merge conflict

kangqiao322 committed:
Merge branch 'master' of https://github.com/kangqiao322/ip

kangqiao322 committed:
Level-2

kangqiao322 committed:
Level-1, Greet, Echo, Exit

kangqiao322 committed:
Level-2, Add, List

kangqiao322 committed:
Level-2, Add, List

kangqiao322 committed:
Level-3, Mark as Done

kangqiao322 committed:
Level-4, ToDos, Events, Deadlines

kangqiao322 committed:
A-TextUiTesting

kangqiao322 committed:
Level-5, Handle Errors

kangqiao322 committed:
Level-6, Delete

kangqiao322 committed:
Level-6, Delete

kangqiao322 committed:
Level-6, Delete

kangqiao322 committed:
Level-6, Delete

kangqiao322 committed:
Level-6, Delete

kangqiao322 committed:
Level 7

kangqiao322 committed:
Level-7

kangqiao322 committed:
Level-8

kangqiao322 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java
#	src/out/production/ip/Deadline.class
#	src/out/production/ip/Duke.class
#	src/out/production/ip/Event.class
#	src/out/production/ip/Task.class
#	src/out/production/ip/ToDo.class

kangqiao322 committed:
Level-8

kangqiao322 committed:
A-MoreOOP

kangqiao322 committed:
A-Packages

kangqiao322 committed:
Merge branch 'nus-cs2103-AY2223S1:add-gradle-support' into add-gradle-support

kangqiao322 committed:
Merge branch 'add-gradle-support'

kangqiao322 committed:
A-Gradle

kangqiao322 committed:
A-Gradle

kangqiao322 committed:
A-Gradle

kangqiao322 committed:
A-Gradle

kangqiao322 committed:
Merge branch 'add-gradle-support'

kangqiao322 committed:
Merge branch 'master' of https://github.com/kangqiao322/ip

kangqiao322 committed:
A-Gradle

JovonLim commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    /**
+     * Set the isDone to true
```
I think this should be written as Sets instead? I noticed the same issue in the method directly below.

kangqiao322 replied:
Noted! Thank you for pointing that out.

JovonLim commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,88 @@
+import java.util.Scanner;
+
+
+public class Storage {
```
Perhaps Javadocs should be written for this class and its methods since they are public and are not getters/setters? I noticed a similar issue for the Ui class

JovonLim commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,54 @@
+               storage.readAndSaveFile(dukeFile, taskList);
+                System.out.println("updated file");
+            }
+        }catch (IOException e) {
```
There might be an indentation issue for the try catch here?

JovonLim commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,113 @@
+        taskList.removeTask(index);
+    }
+
+    public void parseInstruction(Storage storage, TaskList taskList) throws DukeException {
```
I like how u split up the instructions to reduce the amount of lines in one method 👍

kangqiao322 replied:
Thank you!

JovonLim commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,48 @@
+        super(description);
+        this.at = at;
+
+        try {
```
Perhaps these try catch blocks for deadline/event could be written in the parser class instead, ensuring only valid inputs for the constructors?
Additionally the class fields description and isDone above may be redundant since you are using super?

kangqiao322 replied:
That does make sense.

JovonLim reviewed (commented):
Overall, I like that most of the comments in your code helps for easier understanding of what the code does as well as identifying current problems of your code. I think its good that you acknowledged them 😄 Just take note of the few violations of the Java coding standard and work on improving those problems!

kangqiao322 committed:
A-JUnit

kangqiao322 committed:
A-JUnit

kangqiao322 committed:
A-JUnit

kangqiao322 committed:
A-JUnit

kangqiao322 committed:
merge

kangqiao322 committed:
A-Jar

kangqiao322 committed:
A-Jar

kangqiao322 committed:
A-jar

kangqiao322 committed:
A-JavaDoc

kangqiao322 committed:
A-CodingStandard

kangqiao322 committed:
Level-9

kangqiao322 committed:
merge JavaDocs

kangqiao322 committed:
merge Level-9

kangqiao322 committed:
Level-9

WingKei67 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+        try{
+            File dukeFile = new File("data/duke.txt");
+            dukeFile.getParentFile().mkdirs();//create the directory
+            if(dukeFile.createNewFile()){
```
Would it be better if you add one whitespace after if?

WingKei67 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+     */
+    public void run() throws DukeException {
+        System.out.println(ui.getHello());
+        try{
```
Would it be better if you add one whitespace after try?

WingKei67 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,148 @@
+
+    private static void parseFind(String str, TaskList taskList) throws DukeException {
+        Ui ui = new Ui(taskList);
+        String find = str.split(" ")[1].trim();
```
Is it better to name a String variable using a noun?

WingKei67 commented on src/main/java/duke/ToDo.java:
```diff
@@ -0,0 +1,29 @@
+ * Inherits from the Task class.
+ *
+ */
+public class   ToDo extends Task {
```
Is there something wrong with the space here?

WingKei67 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,59 @@
+                this.localDate = LocalDate.parse(date.trim(),
+                        DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+            } catch (DateTimeParseException e1) {
+                System.out.println("Please provide the date in the " +
```
Would it be better if the exception message can be handled by UI class?

WingKei67 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+package duke;
```
Is it better to split the files into different sub packages?

WingKei67 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,104 @@
+                inputs.addTask(newTask);
+            }
+        }
+    }
```
I really appreciate how you organize if-else blocks and for loop with good coding styles.

kangqiao322 committed:
Level-10

kangqiao322 committed:
Level-10

kangqiao322 committed:
A-Assertions

kangqiao322 committed:
mainWindow class: added assertions at the userImages and dukeImages
variables.

It is to document important assumptions at these important points that
should hold in the code.

Assertions are added around those important points to verify that these
variables are never null.

Assertions are placed at those points to indicate possible bugs at those
points.

kangqiao322 committed:
Merge pull request #1 from kangqiao322/branch-A-Assertions

Branch a assertions

kangqiao322 committed:
A-CodeQuality

kangqiao322 committed:
Parser class: added a try-catch block to handle the IndexOutOfBounds excpetion
in the parseDelete function when the user attempts to delete a task
with an index that does not exist in the list.

The user will not be able to know why he/she is unable to
input the command when this exception is not handled.

A try-catch block is added to catch the exception when it is
encountered and a message telling the user what is
the problem for him/she to diagnose the issue.

A try-catch block is the easiest and simplest way to handle such
exceptions

kangqiao322 committed:
Merge pull request #2 from kangqiao322/branch-A-CodeQuality

Branch a code quality

kangqiao322 committed:
C-DetectDuplicates

kangqiao322 committed:
Changes from feedback

kangqiao322 committed:
Ui.png

kangqiao322 committed:
removed kra file

kangqiao322 committed:
A-UserGuide

kangqiao322 committed:
Update README.md

kangqiao322 committed:
Update README.md

kangqiao322 committed:
Update README.md

kangqiao322 committed:
fixed bug on duplicated tasks being added to the list

kangqiao322 committed:
Minor bug fix for wrong reply for incorrect event input

kangqiao322 committed:
Update README.md

kangqiao322 committed:
Update README.md

kangqiao322 committed:
Update README.md
